include/seastar/http/client.hh

```diff
@@ -94,8 +94,14 @@ private:
     future<reply> recv_reply();
 };
 
+class connection_factory {
+public:
+    virtual future<connected_socket> make() = 0;
+    virtual ~connection_factory() {}
+};
+
```

Comment:
It's good that the factory is an interface, it allows the caller to set whatever policies they want.

Reply:
But, it needs documentation.